### PR TITLE
Document Spark types

### PR DESCRIPTION
**Pull request by zhli1142015**

### CODE REVIEW

File: velox/docs/develop/types.rst [hunk 1]
```diff
@@ -145,3 +145,38 @@ Supported range of milliseconds is [0xFFF8000000000000L, 0x7FFFFFFFFFFFF]
 (or [-69387-04-22T03:45:14.752, 73326-09-11T20:14:45.247]). The low 12 bits
 store timezone ID. Supported range of timezone ID is [1, 1680].
 The definition of timezone IDs can be found in ``TimeZoneDatabase.cpp``.
+
+Spark Types
+~~~~~~~~~~~~
+The data types in Spark have some semantic differences compared to those in
```
Comment (mbasmanova) [inline velox/docs/develop/types.rst:151]: perhaps, add a link to Spark's documentation: https://spark.apache.org/docs/latest/sql-ref-datatypes.html

File: velox/docs/develop/types.rst [hunk 2]
```diff
@@ -145,3 +145,38 @@ Supported range of milliseconds is [0xFFF8000000000000L, 0x7FFFFFFFFFFFF]
...
+
+      SELECT min(a)
+      FROM (
+          VALUES
+              (ARRAY[1, 2]),
+              (ARRAY[1, null])  
+      ) AS t(a);
+      -- ARRAY[1, null]
+
+* Map type is not comparable in Spark. Map type is comparable in Presto if both
```
Comment (mbasmanova) [inline velox/docs/develop/types.rst:181]: nit: perhaps, clarify some more

```
MAP type is not comparable and not orderable in Spark. In Presto, MAP type is also not orderable, but it is comparable if both key and value types are comparable. The implication is that MAP type cannot be used as a join, group by or order by key in Spark.
```

**Review by mbasmanova — APPROVED** [review]
@zhli1142015 Thank you for improving the docs. Looks great % couple of nits.

@rui-mo Rui, would you also take a look?

File: velox/docs/develop/types.rst (hunk 1 above)
Comment (zhli1142015) [reply velox/docs/develop/types.rst]: Updated, thanks.

**Review by mbasmanova — APPROVED** [review]
Thanks.

File: velox/docs/develop/types.rst [hunk 3]
```diff
@@ -145,3 +145,40 @@ Supported range of milliseconds is [0xFFF8000000000000L, 0x7FFFFFFFFFFFF]
 (or [-69387-04-22T03:45:14.752, 73326-09-11T20:14:45.247]). The low 12 bits
 store timezone ID. Supported range of timezone ID is [1, 1680].
 The definition of timezone IDs can be found in ``TimeZoneDatabase.cpp``.
+
+Spark Types
+~~~~~~~~~~~~
```
Comment (rui-mo) [inline velox/docs/develop/types.rst:150]: nit: remove one `~` to align the line length with `Spark Types`.

File: velox/docs/develop/types.rst [hunk 4]
```diff
@@ -145,3 +145,40 @@ Supported range of milliseconds is [0xFFF8000000000000L, 0x7FFFFFFFFFFFF]
...
+          VALUES
+              (ARRAY[1, 2]),
+              (ARRAY[1, null])  
+      ) AS t(a);
+      -- ARRAY[1, null]
+
+* MAP type is not comparable and not orderable in Spark. In Presto, MAP type is
+  also not orderable, but it is comparable if both key and value types are
+  comparable. The implication is that MAP type cannot be used as a join, group
+  by or order by key in Spark.
```
Comment (rui-mo) [inline velox/docs/develop/types.rst:184]: Perhaps also mention window partition key.

**Review by rui-mo — COMMENTED** [review]
Thanks.

**Pull request closed without merging**